File: mobile/android-native/app/src/main/kotlin/ai/tasktrooper/androidnative/data/Task.kt
package ai.tasktrooper.androidnative.data

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// Matches the Task shape every backend boilerplate in this repo exposes.
@Serializable
data class Task(
    val id: String,
    val title: String,
    val done: Boolean,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String,
)

@Serializable
data class CreateTaskRequest(val title: String)

@Serializable
data class SetDoneRequest(val done: Boolean)

File: mobile/flutter/lib/models/task.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'task.freezed.dart';
part 'task.g.dart';

/// Matches the Task shape every backend boilerplate in this repo
/// exposes (go-fiber/java-spring/quarkus REST /tasks).
@freezed
abstract class Task with _$Task {
  const factory Task({
    required String id,
    required String title,
    required bool done,
    @JsonKey(name: 'created_at') required DateTime createdAt,
    @JsonKey(name: 'updated_at') required DateTime updatedAt,
  }) = _Task;

  factory Task.fromJson(Map<String, dynamic> json) => _$TaskFromJson(json);
}
